File: app/src/main/java/org/sco/movieratings/data/MovieContract.java
package org.sco.movieratings.data;

import android.content.ContentResolver;
import android.content.ContentUris;
import android.net.Uri;
import android.provider.BaseColumns;

public class MovieContract {

    public static final String CONTENT_AUTHORITY = "org.sco.movieratings";

    public static final Uri BASE_CONTENT_URI = Uri.parse("content://" + CONTENT_AUTHORITY);

    public static final String PATH_MOVIES = "movies";

    public static final class MovieEntry implements BaseColumns {

        public static final Uri CONTENT_URI =
                BASE_CONTENT_URI.buildUpon().appendPath(PATH_MOVIES).build();

        public static final String CONTENT_TYPE =
                ContentResolver.CURSOR_DIR_BASE_TYPE + "/" + CONTENT_AUTHORITY + "/" + PATH_MOVIES;
        public static final String CONTENT_ITEM_TYPE =
                ContentResolver.CURSOR_ITEM_BASE_TYPE + "/" + CONTENT_AUTHORITY + "/" + PATH_MOVIES;

        // Table name
        public static final String TABLE_NAME = "movies";

        // ID Returned by the API
        public static final String COLUMN_MOVIE_ID = "movie_id";

        public static Uri buildMovieUri(long id) {
            return ContentUris.withAppendedId(CONTENT_URI, id);
        }

        public static Uri buildMovieId(long id) {
            return ContentUris.withAppendedId(CONTENT_URI, id);
//            return CONTENT_URI.buildUpon().appendPath(movieId).build();
        }

    }

}

File: app/src/main/java/org/sco/movieratings/db/MovieContract.kt
package org.sco.movieratings.db

import android.content.ContentUris
import android.net.Uri
import android.provider.BaseColumns

object MovieContract {

    internal const val TABLE_NAME = "movie"

    @kotlin.jvm.JvmField
    val CONTENT_URI: Uri = Uri.withAppendedPath(CONTENT_AUTHORITY_URI, TABLE_NAME)
    const val CONTENT_TYPE = "vnd.android.cursor.dir/vnd.$CONTENT_AUTHORITY.$TABLE_NAME"
    const val CONTENT_ITEM_TYPE = "vnd.android.cursor.item/vnd.$CONTENT_AUTHORITY.$TABLE_NAME"

    object MovieEntry : BaseColumns {
        const val MOVIE_ID = "movie_id"
        const val MOVIE_TITLE = "movie_title"
        const val OVERVIEW = "overview"
        const val POSTER_PATH = "poster_path"
        const val RELEASE_DATE = "release_date"
        const val RATING = "vote_average"
        const val POPULARITY = "popularity"
        const val IS_FAVORITE = "is_favorite"
    }

    fun buildUriFromId(id: Long): Uri {
        return ContentUris.withAppendedId(CONTENT_URI, id)
    }
}